app/src/main/java/com/example/myapp/Fragments/FragmentList.kt

```diff
@@ -0,0 +1,117 @@
+import retrofit2.Call
+import retrofit2.Callback
+import retrofit2.HttpException
+import retrofit2.Response
+
+
+class FragmentList() : Fragment(R.layout.fragment_list) {
+
+    private val binding: FragmentListBinding by viewBinding()
+
+    val adapter by lazy {
+        CharacterAdapter(requireContext(), userClick = {
+            findNavController().navigate(FragmentListDirections.toDetailsFragment(it.id))
+        })
+    }
+
+    private val charactersToolbarSearch = mutableListOf<Character>()
+    private var currentRequest: Call<List<Character>>? = null
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        executeRequest()
+
+        with(binding) {
+            toolbar
+                .menu
+                .findItem(R.id.action_search)
+                .actionView
+                .let { it as SearchView }
+                .setOnQueryTextListener(object : SearchView.OnQueryTextListener {
+                    override fun onQueryTextSubmit(p0: String): Boolean {
+                        return false
+                    }
+
+                    override fun onQueryTextChange(p0: String): Boolean {
+                        val list = charactersToolbarSearch.filter { it.name.contains(p0) }
+                            .map { PagingData.Item(it) }
+                        adapter.submitList(list)
+                        return true
+                    }
+                })
+
+            recyclerView.adapter = adapter
+            recyclerView.addItemDecoration(
+                MaterialDividerItemDecoration(
+                    requireContext(),
+                    MaterialDividerItemDecoration.VERTICAL
+                )
+            )
+
+            swipeRefresh.setOnRefreshListener {
+                executeRequest {
+                    swipeRefresh.isRefreshing = false
+                }
+            }
+            executeRequest()
+        }
+    }
+
+    private fun executeRequest(onRequestFinished: () -> Unit = {}) {
+        val finishRequest = {
+            onRequestFinished()
+            currentRequest = null
+        }
+
+        currentRequest?.cancel()
+        currentRequest = Service.api
+            .getCharacters()
+            .apply {
+                enqueue(object : Callback<List<Character>> {
+                    override fun onResponse(
+                        call: Call<List<Character>>,
+                        response: Response<List<Character>>
+                    ) {
+                        if (response.isSuccessful) {
+                            val characters = response.body() ?: return
+                            charactersToolbarSearch.addAll(characters)
+                            val items = characters.map { PagingData.Item(it) } + PagingData.Loading
+                            adapter.submitList(items)
+                        } else {
+                            HttpException(response).message()
```

Comment:
ты на этой строчке создаешь объект ошибки и вызываешь метод для получения строки, дальше ничего не происходит, равносильно тому, что ты просто создал в коде строку и никуда не присвоил
```suggestion
                            handleError(HttpException(response))
```